### ISSUE TITLE
Improved Vanilla not working in 4.7.8

### ISSUE THREAD

**Issue by sapphirecat** [FiendsOfTheElements/FF1Randomizer#1052]
Loaded preset "Improved Vanilla", uploaded ROM (same one as always), clicked Generate, got exception:

```
Progress: 14/25

Placing Out of Bound Bat

System.ArgumentNullException: ArgumentNull_Generic Arg_ParamName_Name, source
   at System.Linq.ThrowHelper.ThrowArgumentNullException(ExceptionArgument )
   at System.Linq.Enumerable.TryGetFirst[IRewardSource](IEnumerable`1 , Func`2 , Boolean& )
   at System.Linq.Enumerable.First[IRewardSource](IEnumerable`1 , Func`2 )
   at FF1Lib.ShopData.UpdateShopSlotPlacement(List`1 placement)
   at FF1Lib.FF1Rom.Randomize(Blob seed, Flags flags, Preferences preferences)
   at FF1Blazorizer.Tabs.FileTab.DoRandomize()

Generating Final Fantasy Randomizer ROM ... Please Wait ... FAILURE: ArgumentNull_Generic Arg_ParamName_Name, source
```

Observed with seed F5574F3D and flags: If-vmh0k22Mu5RhFD3kkBJbHZjiPkfDZX9Z-YXY656J6aex9suHkvxE-PES30UbflTt6iCg940RHVW9sXu93DoE8ghAtpLvIOVx6C9y4f4mRBJ1K0UZtVGKrE99Yx-oQk-nDjZl6vF2FIbDsUN7nggqF6s.r2vmfR2XXtgg

AFAICT, it is independent of my Fun%/Quality-of-Life settings.

**Comment by wildham0** [comment]
Thanks for reporting the issue, it will be fixed in the next release, in the meantime you can use 4.7.7 to play Improved Vanilla
https://4-7-7.finalfantasyrandomizer.com/

**Comment by wildham0** [comment]
Fixed with 4.7.9
